src/main/java/bisq/seednode/SeedNodeMain.java

```diff
@@ -17,181 +17,56 @@
-    private volatile boolean stopped;
-    private static long maxMemory = MAX_MEMORY_MB_DEFAULT;
-
-    static {
-        // Need to set default locale initially otherwise we get problems at non-english OS
-        Locale.setDefault(new Locale("en", Locale.getDefault().getCountry()));
-
-        Utilities.removeCryptographyRestrictions();
-    }
 
     public static void main(String[] args) throws Exception {
-        final ThreadFactory threadFactory = new ThreadFactoryBuilder()
-                .setNameFormat("SeedNodeMain")
-                .setDaemon(true)
-                .build();
-        UserThread.setExecutor(Executors.newSingleThreadExecutor(threadFactory));
-
         // We don't want to do the full argument parsing here as that might easily change in update versions
         // So we only handle the absolute minimum which is APP_NAME, APP_DATA_DIR_KEY and USER_DATA_DIR
         BisqEnvironment.setDefaultAppName("bisq_seednode");
-        OptionParser parser = new OptionParser();
-        parser.allowsUnrecognizedOptions();
-        parser.accepts(AppOptionKeys.USER_DATA_DIR_KEY, description("User data directory", DEFAULT_USER_DATA_DIR))
-                .withRequiredArg();
-        parser.accepts(AppOptionKeys.APP_NAME_KEY, description("Application name", DEFAULT_APP_NAME))
-                .withRequiredArg();
 
-        OptionSet options;
-        try {
-            options = parser.parse(args);
-        } catch (OptionException ex) {
-            System.out.println("error: " + ex.getMessage());
-            System.out.println();
-            parser.printHelpOn(System.out);
-            System.exit(EXIT_FAILURE);
-            return;
-        }
-        BisqEnvironment bisqEnvironment = getBisqEnvironment(options);
+        if (BisqExecutable.setupInitialOptionParser(args)) {
+            // For some reason the JavaFX launch process results in us losing the thread context class loader: reset it.
+            // In order to work around a bug in JavaFX 8u25 and below, you must include the following code as the first line of your realMain method:
+            Thread.currentThread().setContextClassLoader(SeedNodeMain.class.getClassLoader());
 
-        // need to call that before BisqAppMain().execute(args)
-        BisqExecutable.initAppDir(bisqEnvironment.getProperty(AppOptionKeys.APP_DATA_DIR_KEY));
+            new SeedNodeMain().execute(args);
+        }
+    }
 
-        // For some reason the JavaFX launch process results in us losing the thread context class loader: reset it.
-        // In order to work around a bug in JavaFX 8u25 and below, you must include the following code as the first line of your realMain method:
-        Thread.currentThread().setContextClassLoader(SeedNodeMain.class.getClassLoader());
+    @Override
+    protected void doExecute(OptionSet options) {
+        super.doExecute(options);
 
-        new SeedNodeMain().execute(args);
+        checkMemory(bisqEnvironment, seedNode);
```

Comment:
`doExecute()` from parent class will block forever with `keepRunning()`.

Reply:
Ah good catch ;-). Will push a fix commit in a minute...